### DISCUSSION TITLE
How can I get mesh metrics such as "Face alignment", "Skewness", "Volume" information by script in AEDT Icepak?

### DISCUSSION THREAD

**Discussion by vvvvnotwww** [ansys/pyaedt#5327] [Q&A] [answered]
I want to get mesh metrics information in AEDT Icepak. Could anyone tell me if there is any method to get it? Thank you in advance!

**Comment by vvvvnotwww** [comment]
I want to use "export_mesh_stats" to get it, but it has an error.
PyAEDT ERROR: AEDT GRPC API CALL ERROR on export_mesh_stats
PyAEDT ERROR: Failed to execute grpc AEDT command: ExportMeshStats
PyAEDT ERROR:   File "C:\Users\hew8szh\AppData\Local\Temp\ipykernel_45948\4071780270.py", line 1, in <module>
PyAEDT ERROR:     icepak_design.export_mesh_stats(setup_name="Setup1")
PyAEDT ERROR:   File "C:\Users\hew8szh\AppData\Roaming\Python\Python39\site-packages\pyaedt\application\Analysis3D.py", line 292, in export_mesh_stats
PyAEDT ERROR: Method arguments: 
PyAEDT ERROR:     setup_name = Setup1 
PyAEDT ERROR: Check Online documentation on: https://aedt.docs.pyansys.com/version/stable/search.html?q=export_mesh_stats

**Reply by Samuelopez-ansys** [reply]
Hi  @vvvvnotwww ,

This method is only valid for HFSS and Maxwell.

In Icepak there is not an equivalent method.

I found in the AEDT API documentation this method:

![image](https://github.com/user-attachments/assets/706f04db-5276-465b-9d03-934a88bae709)

Please if it does what you need, submit an Issue.

@lorenzovecchietti Any additional idea?

**Comment by lorenzovecchietti** [comment]
@Samuelopez-ansys  I checked your suggestion and the method seems failing in Icepak: `"[error] Script macro error: Surface mesh summary data is not supported (05:47:47 PM  Oct 22, 2024)"`. I'll double-check with development. 

I suspect we have no method in Icepak to achieve that. But I'll investigate in the next few days.

**Reply by vvvvnotwww** [reply]
Hello @Samuelopez-ansys & @lorenzovecchietti , thanks a lot for your help! I want to get mesh metrics because sometimes it can't satisfy minimum requirement to start solve, so I want to get it to judge if I need improve mesh quality. I will also investigate it, if you have any updates, thanks a lot for your sharing!

**Comment by lorenzovecchietti** [comment]
@vvvvnotwww Hi, what you want to achieve is not currently possible, but we can create an enhancement request to have a native API. Can you reach out to me directly at [...] so that we can very briefly discuss this?

**Reply by vvvvnotwww** [reply]
Hello @lorenzovecchietti , sorry for late reply, Do you have time at next Monday 4:30-5:00 p.m. (UTC+08:00)? We can discuss it briefly. I will send an email to you, if the time is not suitable, we can change it. Thanks you in advance!

**Comment by lorenzovecchietti** [comment, marked as answer]
I submitted an enhancement request to obtain meshing data via API. Currently, the only solution is to attempt running the analysis and check if the solver adds error messages to the log. If that happens, you can implement custom logic to handle it.

**Reply by vvvvnotwww** [reply]
Thanks a lot for your help! @lorenzovecchietti It is the best way to solve this question now.😊
